leaderelection/src/main/java/io/scalecube/leaderelection/RaftStateMachine.java

```diff
@@ -0,0 +1,58 @@
+package io.scalecube.leaderelection;
+
+
+import unquietcode.tools.esm.EnumStateMachine;
+
+/**
+ * RAFT state machine dealing with state transition FOLLOWER, CANNIDATE, LEADER.
+ * 
+ * @author Ronen Nachmias
+ */
+public abstract class RaftStateMachine {
```

Comment:
Also it's not clear why it's abstract class? Is it supposed to be extended by some other classes except LeaderElection? If not then embed it into leader election.